BTrees/_base.py

```diff
@@ -1494,3 +1546,44 @@ def MERGE_WEIGHT_default(self, value, weight):
 
 def MERGE_WEIGHT_numeric(self, value, weight):
     return value * weight
+
+def _fix_pickle(mod_dict, mod_name):
+    # Make the pure-Python objects pickle with the same
+    # class names and types as the C extensions by setting the appropriate
+    # _BTree_reduce_as attribute.
+    # If the C extensions are not available, we also change the
+    # __name__ attribute of the type to match the C name (otherwise
+    # we wind up with *Py in the pickles)
+    # Each module must call this as `_fix_pickle(globals(), __name__)`
+    # at the bottom.
+
+    mod_prefix = mod_name.split('.')[-1][:2] # BTrees.OOBTree -> 'OO'
+    bucket_name = mod_prefix + 'Bucket'
+    py_bucket_name = bucket_name + 'Py'
+
+    have_c_extensions = mod_dict[bucket_name] is not mod_dict[py_bucket_name]
+
+    for name in 'Bucket', 'Set', 'BTree', 'TreeSet', 'TreeIterator':
+        raw_name = mod_prefix + name
+        py_name = raw_name + 'Py'
+        try:
+            py_type = mod_dict[py_name]
+        except KeyError:
+            if name == 'TreeIterator':
+                # Optional
+                break
```

Comment:
I'd rather use `continue` here, even though there's no difference in behavior as long as `TreeIterator` is last in the list of things to iterate over.